### PR TITLE
[IMP] More user friendly access error messages

### PR DESCRIPTION
**Pull request by sebgeelen**
* Improve layout of access error messages from group restrictions
* Improve layout of access error messages from security records rules
* Use the error title in the JS crash manager when possible

task id : 2206787

### CODE REVIEW

File: odoo/addons/base/models/ir_rule.py [hunk 1]
```diff
@@ -212,36 +212,47 @@ def _make_access_error(self, operation, records):
...
-        # clean up the cache of records prefetched with display_name above
-        for record in records[:6]:
-            record._cache.clear()
-        return error
+            'document_model': model
+        }
+        msg_footer = _("Contact your administrator to request access if necessary.")
+        msg = '%s \n\n%s' % (msg_heads[operation], msg_footer)
+
+        if self.env.user.has_group('base.group_no_one'):
```
Comment (fmdl) [inline odoo/addons/base/models/ir_rule.py:229]: **if self.env.user.has_group('base.group_no_one') and self.env.user.has_group('base.group_user'):**

@odony 

if case one day portal can be in base.group_no_one.

**Comment by mart-e** [comment]
Hi!
Could you integrate the parameters to benefit from the [new gettext syntax](https://github.com/odoo/odoo/pull/52155) `_("foo %(bar)s" bar=42)` ?
Thanks

**Comment by tivisse** [comment]
To validate: Don't we expose to much information to external users (portal/public).

File: odoo/addons/base/models/ir_model.py [hunk 2]
```diff
@@ -1718,26 +1718,28 @@ def check(self, model, mode='read', raise_exception=True):
...
-            }
-            msg_params = {
-                'document_kind': self.env['ir.model']._get(model).name or model,
-                'document_model': model,
+                'read': _("You are not allowed to access '%(document_kind)s' (%(document_model)s) records."),
+                'write':  _("You are not allowed to modify '%(document_kind)s' (%(document_model)s) records."),
+                'create': _("You are not allowed to create '%(document_kind)s' (%(document_model)s) records."),
+                'unlink': _("You are not allowed to delete '%(document_kind)s' (%(document_model)s) records."),
             }
+            msg_heads = _(
```
Comment (tivisse) [inline odoo/addons/base/models/ir_model.py:1728]: This is not correct, and would have been detected by: https://github.com/odoo/odoo/pull/53683

Comment (sebgeelen) [reply odoo/addons/base/models/ir_model.py:1728]: So, I should str() cast the parameters ?

Comment (mart-e) [reply odoo/addons/base/models/ir_model.py:1728]: @sebgeelen see https://www.odoo.com/documentation/master/reference/translations.html#variables and https://www.odoo.com/documentation/master/reference/guidelines.html#use-translation-method-correctly for more examples. The first argument of the `_` method should always be a plain string, never a variable or an operation (this is what the pylint test added at #53683 will check)

**Comment by sebgeelen** [comment]
> To validate: Don't we expose to much information to external users (portal/public).
@tivisse 

cf task : 
> NB: As YTI pointed out in his note, this spec considers the user being denied access in an internal user in debug mode.

I don't think my changes modify the kind of information provided to the user. It's mainly formatting and phrasing.

File: odoo/addons/base/models/ir_model.py (hunk 2 above)
Comment (sebgeelen) [reply odoo/addons/base/models/ir_model.py:1728]: ok thanks, will check

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**